base/multidimensional.jl

```diff
@@ -21,27 +21,17 @@ end
 
 CartesianIndex{N}(index::NTuple{N,Integer}) = CartesianIndex{N}(index)
 (::Type{CartesianIndex})(index::Integer...) = CartesianIndex(index)
-(::Type{CartesianIndex{N}}){N}(index::Integer...) = CartesianIndex{N}(index)
+(::Type{CartesianIndex{N}}){N}(index::Vararg{Integer,N}) = CartesianIndex{N}(index)
 # Allow passing tuples smaller than N
-@generated function (::Type{CartesianIndex{N}}){N,M}(index::NTuple{M,Integer})
-    M == N && return :(CartesianIndex(index))
-    M > N && return :(throw(DimensionMismatch("Cannot create CartesianIndex{$N} from $M indexes")))
-    args = [i <= M ? :(index[$i]) : 1 for i = 1:N]
-    :(CartesianIndex(tuple($(args...))))
-end
+(::Type{CartesianIndex{N}}){N}(index::Tuple) = CartesianIndex{N}(fill_to_length(index, 1, Val{N}))
+(::Type{CartesianIndex{N}}){N}(index::Integer...) = CartesianIndex{N}(index)
```

Comment:
I know this is already merged but only got to look at it now. Couldn't line 27 be merged with line 24 into a single definition?